src/cmd-remote-build-container

```diff
@@ -0,0 +1,130 @@
+    '''
+    if output:
+        try:
+            output = check_output(command.split())
+        except CalledProcessError as e:
+            raise SystemExit(f"Failed to invoke command: {e}")
+        return output
+    else:
+        try:
+            check_call(command.split())
+        except CalledProcessError as e:
+            raise SystemExit(f"Failed to invoke command: {e}")
+
+def build(expire, gitURL, gitRef, repo, tag):
+    '''
+    Build the image  using podman remote and push to the registry
+    @param repo str registry repository
+    @param tag  str image tag
+    @param time str expiration time for Quay images
+    '''
+    if expire:
+        run_cmd(f"podman --remote build {gitURL}#{gitRef} --tag {repo}:{tag} --label=quay.expires-after={expire}")
+    else:
+        run_cmd(f"podman --remote build {gitURL}#{gitRef} --tag {repo}:{tag}")
+
+def push(repo, tag, force=None):
+    '''
+    Push image to registry
+    @param repo str registry repository
+    @param tag str image tag
+    '''
+    if (search(repo, tag) is True and force is True) or search(repo, tag) is False:
+        run_cmd(f"podman --remote push {repo}:{tag}")
+        # Even though check_call only returns after the cmd completion,
+        # Quay seems to take more time to publish images in some occasions
+        run_cmd("sleep 50")
+        if search(repo, tag) is True:
+            print(f"Build and Push done successfully via tag: {tag}")
+        else:
+            raise SystemExit(f"Registry doesn't seem to have tag:{tag}")
+    else:
+        raise SystemExit(f"Registry tag:{tag} found in {repo}. If you want to overwride it use --force")
+
+def search(repo, tag):
+    '''
+    Search for a tag in the registry
+    @param force str ignore if image is found
+    @param repo str registry repository
+    @param tag  str image tag
+    '''
+    # Podman remote doesn't allow push using digestfile. That's why the tag check is done
+    tags = run_cmd(f"podman search --list-tags {repo}", True)
+    if (tag in str(tags)):
+        return True
+    return False
+
+def main():
+    parse_args()
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        prog="CoreOS Assembler Remote Build",
+        description="Build coreos-assembler remotely",
+        usage="""
+Run multi-arch builds using podman remote.
+In order to get cmd-remote-build-container working the CONTAINER_SSHKEY and CONTAINER_HOST environment variables
+must be defined
+
+Examples:
+    $ cmd-remote-build-container \
+        --arch aarch64 \
+        --expire 4d \
+        --git-ref main \
+        --git-url https://github.com/coreos/coreos-assembler.git \
+        --repo quay.io/coreos/coreos-assembler-staging \
+        --push-to-registry """)
+
+    parser.add_argument(
+        '--arch', required=True,
+        help='Build Architecture')
+    parser.add_argument(
+        '--expire', required=False,
+        help='Expiration time for Quay images')
```

Comment:
This could be `--label` instead if we wanted to hoist the quay implementation details out of this code. So instead of `--expire 4d` someone would pass `--label quay.expires-after=4d`.

I'm good with it like it is, but if there are new labels we want to take advantage of in the future the other way might be better.

I'm really on the fence about which one I prefer.